### PR TITLE
PRO-1282 check for drafts modified prop

### PR DESCRIPTION
**Pull request by stuartromanek**
https://linear.app/apostrophecms/issue/PRO-1282/pieces-not-getting-the-pending-update-label

@boutell as a side note, its confusing we're passing `item`, `draft`, and `published` into all these components, `item` being a moving target. Should we just normalize all the cells to expect `draft` and `published`?

### CODE REVIEW

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
     class="apos-table__cell-field apos-table__cell-field--labels"
     :class="`apos-table__cell-field--${header.name}`"
   >
-    <span v-if="manuallyPublished && item.modified && !item.submitted && item.lastPublishedAt">
+    <span v-if="manuallyPublished && (item.modified || draft.modified) && !item.submitted && item.lastPublishedAt">
```
Comment (boutell) [inline modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue]: Use `draft && draft.modified`. `item` is for display purposes and isn't guaranteed to know about draft/live matters, it exists so you don't have to worry about that if all you want to do is display "what we think you should display."

**Review by boutell — COMMENTED** [review]
@stuartromanek re: your side note it was agreed in a previous round of code review changes that the "item" passed to cells should be "what we want you to show people," abstracting that decision up a level and out of these cell components, but that we also need "draft" for cases where what we want to show is literally related to draft publication progress.

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue [hunk 2]
```diff
@@ -3,28 +3,28 @@
     class="apos-table__cell-field apos-table__cell-field--labels"
     :class="`apos-table__cell-field--${header.name}`"
   >
-    <span v-if="manuallyPublished && item.modified && !item.submitted && item.lastPublishedAt">
+    <span v-if="manuallyPublished && draft.modified && draft.submitted && draft.lastPublishedAt">
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:6]: Shouldn't the last part still be `item.lastPublishedAt`? It's indicating a pending draft on a published thing.

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue [hunk 3]
```diff
@@ -3,28 +3,28 @@
...
       <AposLabel
         label="Pending Updates" class="apos-table__cell-field__label"
         tooltip="There are active changes to this document."
         :modifiers="[ 'is-success', 'is-filled' ]"
       />
     </span>
-    <span v-else-if="item.submitted">
+    <span v-else-if="draft.submitted">
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:13]: Why do we have this if the previous one also checks for `draft.submitted`?

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue [hunk 4]
```diff
@@ -3,28 +3,28 @@
...
-    <span v-if="manuallyPublished && !item.lastPublishedAt">
+    <span v-if="manuallyPublished && !draft.lastPublishedAt">
       <AposLabel
         label="Draft" class="apos-table__cell-field__label"
         :modifiers="[ 'is-warning', 'is-filled' ]"
         tooltip="This document hasn't been published yet."
       />
     </span>
-    <span v-if="item.archived">
+    <span v-if="draft.archived">
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:27]: I think this should still be `item`. Is there even a draft to pass in if it's archived?

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue [hunk 5]
```diff
@@ -3,28 +3,28 @@
...
-    <span v-else-if="item.submitted">
+    <span v-else-if="draft.submitted">
       <AposLabel
         label="Pending" class="apos-table__cell-field__label"
         tooltip="Changes to this document are awaiting approval by an admin or editor."
         :modifiers="[ 'is-filled' ]"
       />
     </span>
-    <span v-if="manuallyPublished && !item.lastPublishedAt">
+    <span v-if="manuallyPublished && !draft.lastPublishedAt">
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:20]: Seems like an unnecessary change.

**Review by abea — CHANGES_REQUESTED** [review]
`draft` defaults to `null` in the mixin, so should be update there or overridden here.

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue (hunk 2 above)
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:6]: `lastPublishedAt` is an aspect of the publication status, which apostrophe always records on the draft for certain, so it should come from `draft`.

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue (hunk 3 above)
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:13]: Just because it is submitted doesn't mean it's all those other things in that "and" statement. If it isn't, we have a fallback that applies to the case where it is merely submitted.

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue (hunk 4 above)
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:27]: There is always a draft. Archived things only exist as drafts.

Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:27]: (Which means I could check item.archived, technically, but the simplifying principle I'm pushing for here is *if you are checking something relating to the lifecycle status of the document, look at draft.*)

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue (hunk 5 above)
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:20]: We do eventually copy this to `published`, but I believe it is best to get anything relating to publication status from the draft, which is the primary place Apostrophe stores it.

File: modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue (hunk 4 above)
Comment (abea) [reply modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue:27]: oh right.

File: modules/@apostrophecms/ui/ui/apos/mixins/AposCellMixin.js [hunk 6]
```diff
@@ -12,9 +12,7 @@ export default {
     },
     draft: {
       type: Object,
-      default() {
-        return null;
-      }
+      required: true
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/mixins/AposCellMixin.js:15]: Since `draft` is now required, we have no need for `item` as a prop. The manager is doing logic that could be done in the mixin instead. Contain that and let the manager only pass in two clear things.

**Review by abea — CHANGES_REQUESTED** [review]

**Review by boutell — APPROVED** [review]
